src/pip/_internal/locations.py

```diff
@@ -51,22 +51,24 @@ def virtualenv_no_global():
         return False
 
 
-if running_under_virtualenv():
-    src_prefix = os.path.join(sys.prefix, 'src')
-else:
-    # FIXME: keep src in cwd for now (it is not a temporary folder)
-    try:
-        src_prefix = os.path.join(os.getcwd(), 'src')
-    except OSError:
-        # In case the current working directory has been renamed or deleted
-        sys.exit(
-            "The folder you are executing pip from can no longer be found."
-        )
+def get_src_prefix():
+    if running_under_virtualenv():
+        src_prefix = os.path.join(sys.prefix, 'src')
+    else:
+        # FIXME: keep src in cwd for now (it is not a temporary folder)
+        try:
+            src_prefix = os.path.join(os.getcwd(), 'src')
+        except OSError:
+            # In case the current working directory has been renamed or deleted
+            sys.exit(
+                "The folder you are executing pip from can no longer be found."
+            )
+
+    # under macOS + virtualenv sys.prefix is not properly resolved
+    # it is something like /path/to/python/bin/..
+    # Note: using realpath due to tmp dirs on OSX being symlinks
```

Comment:
Any idea why this comment say "using realpath ...", but we're using `abspath`?

Reply:
Good point, the comment should have been removed in https://github.com/pypa/pip/commit/7288475118a8c0b73d58789f740f930c7f6a3938#diff-f21e4e94baf6a2d2f98faab0451e418aL165